lib/Target/SubstraitPB/Import.cpp

```diff
@@ -128,6 +163,66 @@ importExpression(ImplicitLocOpBuilder builder, const Expression &message) {
   }
 }
 
+static mlir::FailureOr<FieldReferenceOp>
+importFieldReference(ImplicitLocOpBuilder builder,
+                     const Expression::FieldReference &message) {
+  using ReferenceSegment = Expression::ReferenceSegment;
+
+  MLIRContext *context = builder.getContext();
+  Location loc = UnknownLoc::get(context);
```

Comment:
Could we do better here?  even just file name

Reply:
I think yes, but this would require some code reorganization. A few notes before I forget: I'd need to implement the translation using `llvm::SourceMgr` and access `BufferBuffer:: getBufferIdentifier` to get the filename.

Reply:
I've created #839 to track this effort.